In [217]:
import pandas as pd

import matplotlib.pyplot as mattyp 


In [218]:


shopping_data = pd.read_csv('shopping.csv')

In [219]:
shopping_data.head()

,Customer ID,Age,Gender,Item Purchased,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Shipping Type,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,3475,NaN,Male,Jacket,30.904467,Maine,M,Burnt orange,Fall,4.0,Standard,No,0,Credit Card,NaN
1,3698,21.0,Female,Backpack,31.588259,NaN,L,Turquoise,Winter,2.0,Express,No,1,Credit Card,Monthly
2,2756,31.0,Male,Leggings,24.231704,Nevada,M,Terra cotta,Winter,4.0,Standard,No,0,Credit Card,NaN
3,3340,NaN,Male,Pajamas,33.918834,Nebraska,M,Black,Winter,NaN,Standard,No,2,Credit Card,Bi-Weekly
4,3391,38.0,Male,Sunglasses,36.545487,Oregon,S,Aubergine,Summer,NaN,Standard,No,0,Credit Card,NaN


In [220]:
shopping_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3900 entries, 0 to 3899
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Customer ID             3900 non-null   int64  
 1   Age                     3510 non-null   float64
 2   Gender                  3900 non-null   object 
 3   Item Purchased          3900 non-null   object 
 4   Purchase Amount (USD)   3900 non-null   float64
 5   Location                3510 non-null   object 
 6   Size                    3900 non-null   object 
 7   Color                   3900 non-null   object 
 8   Season                  3900 non-null   object 
 9   Review Rating           1431 non-null   float64
 10  Shipping Type           3900 non-null   object 
 11  Promo Code Used         3900 non-null   object 
 12  Previous Purchases      3900 non-null   int64  
 13  Payment Method          3900 non-null   object 
 14  Frequency of Purchases  1560 non-null   

In [221]:
shopping_data.describe()

,Customer ID,Age,Purchase Amount (USD),Review Rating,Previous Purchases
count,3900.000000,3510.000000,3900.000000,1431.000000,3900.000000
mean,1950.500000,34.256695,38.708329,3.904962,5.841538
std,1125.977353,10.616340,11.504923,1.105952,5.843695
min,1.000000,18.000000,11.392520,1.000000,0.000000
25%,975.750000,26.000000,29.332617,4.000000,1.000000
50%,1950.500000,33.000000,36.953004,4.000000,4.000000
75%,2925.250000,41.000000,48.739596,5.000000,9.000000
max,3900.000000,75.000000,69.567481,5.000000,20.000000


            Required Analytical Questions
1.  What are the most popular colors by season?

In [222]:
# Most popular color by season
color_counts = shopping_data.groupby(['Season', 'Color']).size().reset_index(name='count')
color_counts = color_counts.sort_values(['Season', 'count'], ascending=[True, False])
top_colors = color_counts.groupby('Season').head(1)
print(top_colors)

    Season      Color  count
4     Fall      Brown    117
18  Spring  Baby blue     83
41  Summer   Lavender    120
53  Winter      Black    155


In [ ]:
: In fall the most popular color is brown with a frequency of 117. 
: In Spring the top color is baby blue with a frequency of 83.
: For Summer lavender with a frequency of 120 and in winter black with a frequency of 155.
: Black was the most popular winter color with a frequency of 155.
: It may be benifical for the company to look into promotions and advertising that corresponds to the most popular color purchases for that particular season. For example promotions for items available in brown during the fall as well as advertising in store on the company website and social media.

2. What is the most popular clothing item by season?

In [223]:
# Most popular clothing item by season
item_counts = shopping_data.groupby(['Season', 'Item Purchased']).size().reset_index(name='count')
item_counts = item_counts.sort_values(['Season', 'count'], ascending=[True, False])
top_items = item_counts.groupby('Season').head(1)
print(top_items)

    Season Item Purchased  count
14    Fall          Socks     87
20  Spring  Running Shoes    106
27  Summer         Shorts    177
35  Winter       Leggings    157


: For fall the most popular item was socks with a frequency of 87.
: For spring running shoes was most popular with a frequency of 106.
: In the Summer shorts was most popular with a frequency 177
: During winter leggings was the most popular item with a frequency of 157.
: It may be benifical for the company to look into promotions and advertising that corresponds to the most popular item purchases for that particular season. For example leggings promotions and advertising during winter or shorts during the summer.

3. What is the effect of promo codes on the dollar amount of purchases?

In [224]:
# Effect of promo codes on purchase amount
promo_effect = shopping_data.groupby('Promo Code Used')['Purchase Amount (USD)'].agg(['count', 'mean', 'median'])
print(promo_effect)

                 count       mean     median
Promo Code Used                             
No                2223  30.172085  30.180773
Yes               1677  50.023814  49.811857


In [ ]:
: For orders where a promo code wasn't used the average amount per order was $30.17.
: For orders where a promo code was used the average amound per order was $50.02. 
: This tells us the promotions were effective in driving revenue.


                    My analytical questions

#1. Using the purchase amount column which gender spent the most money? This can help determine plans for best market outreach strategies.



In [225]:
gender_spend = shopping_data.groupby('Gender')['Purchase Amount (USD)'].agg(['sum', 'mean','count'])

print(gender_spend)

                     sum       mean  count
Gender                                    
Female      73620.710082  38.870491   1894
Male        76032.285805  38.575487   1971
Non-binary   1309.485399  37.413869     35


: Males spent the most overall with a sum of $76,032 and 1971 total purchases made
: Females were not far behind in 2nd place with a total of $73620 spent and 1894 total purchases made
: Non-binary's came in 3rd place with a total of $1309 and 35 total purchases made.
: It should be noted that while males spent the most overall and made the most purchases females actually spent the most per purchases with a an average of $38.8 spent per purchase. This is slightly above males who have an average of 38.5 spent per purchase.
: The data points to males and females being virtually tied for the gender most likely to make a purchase and or the highest value purchase. It would be benefical to lean into this and target these demographics via advertising. We can also think about ways to increase Non-binary reputation in our sales data. Maybe a social media collaboration with a popular Non-binary fashion influencer. 

2. Which season did customers spend the most money? This can help determine when should we run promotions? What time of year should we put the most resources into marketing. 

In [227]:
season_pivot = pd.pivot_table(shopping_data, values = 'Purchase Amount (USD)', index = 'Season', aggfunc=['sum', 'mean', 'count'])

print(season_pivot)

                         sum                  mean                 count
       Purchase Amount (USD) Purchase Amount (USD) Purchase Amount (USD)
Season                                                                  
Fall            41004.570210             38.830085                  1056
Spring          22829.415359             39.293314                   581
Summer          36446.813666             38.690885                   942
Winter          50681.682052             38.366148                  1321


: The season where customers spent the most money in total was winter with a total of $50681.6
: Fall was 2nd with a total purchase amount of 41,004 but actually had a higher average purchase amount per season $38.8 to winters $38.3
: Winter was carried by having the most total purchases for any season 1321. Fall was 2nd with 1056 total purchases.
: It makes sense to invest in advertising and promotions more heavily during winter and fall as sales are up.

3. What is the frequency of purchases by gender for each season?

In [228]:
gender_season = shopping_data.groupby(['Gender', 'Season']).size().reset_index(name='Total Purchases')

print(gender_season)

        Gender  Season  Total Purchases
0       Female    Fall              480
1       Female  Spring              259
2       Female  Summer              500
3       Female  Winter              655
4         Male    Fall              568
5         Male  Spring              318
6         Male  Summer              434
7         Male  Winter              651
8   Non-binary    Fall                8
9   Non-binary  Spring                4
10  Non-binary  Summer                8
11  Non-binary  Winter               15


: Winter has the most total purchases for all genders with Females leading with 655 purchases, Males are 2nd with 651 purchases and Non-binarys are 3rd with 15 total purchases during this timeframe.
: This is more evidnece that it may prove very beneficial to invest in promotion and advertising for leggings during the winter
: For Fall the total purchases were 568 for males, 480 total purchases for females and 8 total purchases for non-binaries
: For Summer the total purchases for females was 500, males 434 and non-binaries made 8 total purchases for fall.
: For Spring males led with 318 total purchases, females were 2nd 259 total purchases and non-binaries made 4 purchases during this timeframe.

5. Out of our most loyal customers people who made purchases 10 or more times what gender are they most likely to belong to? Again this can help with market outreach strategies? Which group should we target for advertising? What form of advertising would be best for us to invest?

In [229]:
loyal = shopping_data[shopping_data['Previous Purchases'] >= 10]

gender_loyal = loyal['Gender'].value_counts().reset_index()
gender_loyal.columns = ['Gender', 'Count']
print(gender_loyal)

       Gender  Count
0      Female    449
1        Male    444
2  Non-binary     10


: Females represents the most likely group of people to purchase 10 or more items currently but this number is almost 50/50 with males 449 to 444 extremely close.
: This gives strong reason to invest in promotion and advertising evenly between these two groups but earlier numbers suggest it's probably a better idea to invest in one group more heavly depending on the season but evenly overall. Males are more likely to purchase during the fall and females are more likely to purchase during the summer. Advertising can reflect this.